src/extra/components/pulses.py

```diff
@@ -1298,6 +1300,14 @@ def __init__(self, data, source=None, instrument=None, *,
             raise ValueError('must specify one of bunch_table_position, '
                              'bunch_table_offset, pulse_offset')
 
+        if self._pulse_offset == 0:
+            # Implement this case via the bunch_table_offset mechanism
+            # to enable it to work even if there is only a single FEL
+            # pulse in any given train.
+            # Implemenmtation-wise the modes are always checked in the
```

Comment:
Nitpicking:
```suggestion
            # Implementation-wise the modes are always checked in the
```

Reply:
Oops, thanks. Also found another typo while fixing it.